src/include/brain/kd_tree.h

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// kd_tree.h
+//
+// Identification: src/include/brain/kd_tree.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "common/macros.h"
+#include "annoy/annoylib.h"
+#include "annoy/kissrandom.h"
+#include "brain/cluster.h"
+
+namespace peloton {
+namespace brain {
+
+//===--------------------------------------------------------------------===//
+// KDTree
+//===--------------------------------------------------------------------===//
+
+class KDTree {
```

Comment:
We should still create a separate .cpp file for this class. The length of the functions in this file exceeds what should be included in the header file. It's not only about reducing compilation dependency, but also about reducing instruction cache misses.